frontend/src/components/IntegrationConfiguration.tsx

```diff
@@ -152,6 +152,11 @@ const MapStates: FC<{ configuration: IntegrationConfiguration }> = ({
             {selected && (
               <SetMapping name={selected.name} select={select(selected.id)} />
             )}
+            {statusMapping && statusMapping.length > 0 && (
```

Comment:
I'd prefer having the whole section under the same condition because multiple conditions may get out of sync and it's not always obvious the conditions are meant to be the same.

```tsx
{selected && (
  <>
    <div ... statusMappingLabel>
    </div>
    <SetMapping name={selected.name} select={select(selected.id)} />
  </>
)}
{columns && statusMapping && statusMapping.length > 0 && (
  <>
    <div ... statusMappingLabel>
    </div>
    {statusMappings.map(...
    ...
  </>
```